### PR TITLE
Create a corresponding Flux OCI Repository for each Snapshot

### PR DESCRIPTION
**Pull request by phoban01**
## Description

This PR creates a Flux OCI source whenever a snapshot is created. The digest is kept in sync with the latest snapshot.

This enables a better automation flow.

Signed-off-by: Piaras Hoban <phoban01@gmail.com>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -36,6 +35,7 @@ var (
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 	utilruntime.Must(deliveryv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(v2beta1.AddToScheme(scheme))
```
Comment (Skarlso) [inline main.go:38]: `autoscaling`? 🤔

Comment (phoban01) [reply main.go:38]: yep just spotted this. have the fix now.

File: config/default/kustomization.yaml [hunk 2]
```diff
@@ -6,3 +6,12 @@ resources:
 - ../rbac
 - ../manager
 - namespace.yaml
+images:
```
Comment (Skarlso) [inline config/default/kustomization.yaml:9]: We should find a way to ignore these or not commit them or something...

File: config/manager/deployment.yaml [hunk 3]
```diff
@@ -29,7 +29,7 @@ spec:
         - --oci-registry-addr=registry.ocm-system.svc.cluster.local:5000
         image: open-component-model/ocm-controller
         name: manager
-        imagePullPolicy: IfNotPresent
+        imagePullPolicy: Always
```
Comment (Skarlso) [inline config/manager/deployment.yaml:32]: Hehehe

Comment (phoban01) [reply config/manager/deployment.yaml:32]: argh...

File: examples/mapping-with-cue/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
```
Comment (Skarlso) [inline examples/mapping-with-cue/kustomization.yaml:1]: Let's not call this kustomization but rather something like sample_kustomization.yaml. As we've seen earlier kustomization.yaml is special.

**Review by Skarlso — APPROVED** [review]
Lgtm. One remark.

File: examples/mapping-with-cue/kustomization.yaml (hunk 4 above)
Comment (phoban01) [reply examples/mapping-with-cue/kustomization.yaml]: excellent point. i've renamed all of the examples.

**Pull request merged**